File: src/node.ts
export class Node {
    key: string;
    value: string;
    left: Node | null;
    right: Node | null;
    height: number;

    constructor(key: string, value: string) {
        this.key = key;
        this.value = value;
        this.height = 1;
        this.left = null;
        this.right = null;
    }
}

File: lib/node.go
package lib

type Node struct {
	key    string
	value  string
	height int
	left   *Node
	right  *Node
}

func NewNode(key string, value string) *Node {
	return &Node{
		key:    key,
		value:  value,
		height: 1,
	}
}
